cornerstonejs/cornerstone3D
# fix(EllipticalROITool) Issue 1294 - Ellipse tool spamming ANNOTATION_MODIFIED events when it is outside volume

### Context

Fix for [Issue 1294](https://github.com/cornerstonejs/cornerstone3D/issues/1294)

The Ellipse tool is generating excessive ANNOTATION_MODIFIED events when it is placed outside the volume boundaries. This behavior is unexpected and leads to performance issues, such as event flooding and potential application slowdowns.

Issue was generated because of lack area and area units for ellipse annotation with in this specific scenario. 

Fixed by allowing calculation of area for ellipse annotation even if it's not in in mage volume.

### Changes & Results

Before: [Issue 1294](https://github.com/cornerstonejs/cornerstone3D/issues/1294) video

After:
https://github.com/cornerstonejs/cornerstone3D/assets/2040590/2884ccf9-d53c-4e6d-a155-d1c7ad5d8387


### Testing

1. Open stackAnnotationTools example: `yarn run example stackAnnotationTools`
2.  Subscribe annotations modified event
```
eventTarget.addEventListener(csToolsEnums.Events.ANNOTATION_MODIFIED,(event)=>{
  console.count(`${event.type}: ${event.detail.annotation.metadata.toolName}`)
})
```
3. Select the Ellipse tool from the toolbar.
4. Draw an ellipse within the volume to confirm normal behavior.
5. Move the ellipse outside the volume boundaries.
### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

- (N/A) My code has been well-documented (function documentation, inline comments,
  etc.)

- (N/A) I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)


#### Public Documentation Updates



- (N/A) The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] "Node version: 16.20.2
- [x] "Browser: Chrome 125.0.6422.113

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
